app/k9mail/src/main/java/com/fsck/k9/account/AccountCreator.kt

```diff
@@ -0,0 +1,60 @@
+package com.fsck.k9.account
+
+import android.content.Context
+import app.k9mail.feature.account.setup.domain.ExternalContract
+import app.k9mail.feature.account.setup.domain.entity.Account
+import com.fsck.k9.Account.FolderMode
+import com.fsck.k9.Core
+import com.fsck.k9.Preferences
+import com.fsck.k9.mailstore.SpecialLocalFoldersCreator
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class AccountCreator(
+    private val accountCreatorHelper: AccountCreatorHelper,
+    private val localFoldersCreator: SpecialLocalFoldersCreator,
+    private val preferences: Preferences,
+    private val context: Context,
+    private val coroutineDispatcher: CoroutineDispatcher = Dispatchers.IO,
+) : ExternalContract.AccountCreator {
+
+    override suspend fun createAccount(account: Account): String {
+        return withContext(coroutineDispatcher) {
+            val newAccount = preferences.newAccount()
+
+            newAccount.email = account.emailAddress
+
+            newAccount.incomingServerSettings = account.incomingServerSettings
+            newAccount.outgoingServerSettings = account.outgoingServerSettings
+
+            newAccount.name = account.options.displayName
+            newAccount.senderName = account.options.accountName
+            if (account.options.emailSignature != null) {
+                newAccount.signature = account.options.emailSignature
+            }
+            newAccount.isNotifyNewMail = account.options.showNotification
+            newAccount.automaticCheckIntervalMinutes = account.options.checkFrequencyInMinutes
+            newAccount.displayCount = account.options.messageDisplayCount
+
+            newAccount.folderPushMode = FolderMode.ALL // TODO is this always ALL?
+            newAccount.deletePolicy =
+                accountCreatorHelper.getDefaultDeletePolicy(newAccount.incomingServerSettings.type)
+            newAccount.chipColor = accountCreatorHelper.pickColor()
+
+            // TODO check this:
+            // DI.get(LocalKeyStoreManager.class)
+            //      .deleteCertificate(mAccount, newHost, newPort, MailServerDirection.OUTGOING)
```

Comment:
Deleting old certificates is only necessary when updating server settings of an existing account. Not when creating a new account.